lib/datadog/appsec/actions_handler/stack_trace_collection.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Datadog
+  module AppSec
+    module ActionsHandler
+      # Module that collects the stack trace into formatted hash
+      module StackTraceCollection
+        module_function
+
+        def collect(max_depth:, top_percent:)
+          locations = (caller_locations || []).reject { |location| location.to_s.include?('lib/datadog') }
```

Comment:
I think it should be converted into mutating to avoid allocation

```suggestion
          locations = (caller_locations || [])
          locations.reject! { |location| location.to_s.include?('lib/datadog') }
```